src/Microsoft.IdentityModel.Logging/LogHelper.cs

```diff
@@ -333,22 +333,22 @@ public static void LogWarning(string message, params object[] args)
                 message = format;
 
             if (IdentityModelEventSource.Logger.IsEnabled(eventLevel, EventKeywords.All))
-                IdentityModelEventSource.Logger.Write(eventLevel, innerException, message);
+                IdentityModelEventSource.Logger.Write(eventLevel, innerException!, message);
 
             EventLogLevel eventLogLevel = EventLevelToEventLogLevel(eventLevel);
             if (Logger.IsEnabled(eventLogLevel))
                 Logger.Log(WriteEntry(eventLogLevel, innerException, message, null));
 
             if (innerException != null) 
                 if (string.IsNullOrEmpty(argumentName))
-                    return (T)Activator.CreateInstance(typeof(T), message, innerException);
+                    return (T)Activator.CreateInstance(typeof(T), message, innerException)!;
```

Comment:
for all of these, Activator can return null though, why are we suppressing that here?

Reply:
I don't think `!` is needed on any of them.

Reply:
@keegan-caruso Yes Activator can return null because it's been handled, but IDE throws error code CS8603-Possible null reference return.

Reply:
should be `return (T?)Activator.CreateInstance(typeof(T), message);`

Reply:
this is still incorrect. We either need to actually check that the object created is not null, or we need to change the return type to T?

Reply:
Is this still an issue @FuPingFranco ?